### ISSUE TITLE
Non-text contrast is not sufficient (almost everywhere)

### ISSUE THREAD

**Issue by knowler** [npmx-dev/npmx.dev#1510]
We’re not sufficiently meeting [Success Criterion 1.4.11 (Level AA) of WCAG 2.2](https://www.w3.org/TR/WCAG22/#non-text-contrast) which states:

> The visual [presentation](https://www.w3.org/TR/WCAG22/#dfn-presentation) of the following have a [contrast ratio](https://www.w3.org/TR/WCAG22/#dfn-contrast-ratio) of at least 3:1 against adjacent color(s):
> 
> <dl><dt>User Interface Components</dt>
>   <dd>
>
>   Visual information required to identify [user interface components](https://www.w3.org/TR/WCAG22/#dfn-user-interface-components) and [states](https://www.w3.org/TR/WCAG22/#dfn-states), except for inactive components or where the appearance of the component is determined by the [user agent](https://www.w3.org/TR/WCAG22/#dfn-user-agents) and not modified by the author;
> 
> </dd>
> <dt>Graphical Objects</dt>
>   <dd>
>
>   Parts of graphics required to understand the content, except when a particular presentation of graphics is [essential](https://www.w3.org/TR/WCAG22/#dfn-essential) to the information being conveyed.
> 
> </dd>

This issue has come up in the audit (#1259) and I wanted to surface it early (and in a little more detail). We’re failing for “User Interface Components” as the visual elements such as borders or backgrounds used to distinguish user interface components from their surrounding visual context or to indicate a particular state is not sufficient.

Eric Eggert has a very good blog post on both how to interpret this success criterion and how to practically meet this requirement: [“In detail: 1.4.11 Non-Text Contrast (User Interface Components)”](https://yatil.net/blog/non-text-contrast-in-detail-ui-components).

## What’s affected (work-in-progress)

My audit has only covered the home page so far, but I do know from glancing at other pages, they are affected. I’ll fill out this list as I find specific changes we need to make.

- [ ] Hover state for the “compare” and “settings” links, as well as, the “connect” menu button.
  - Dark theme contrast ratio: 1.274:1 (`#272727` / `#101010`)
  - Light theme contrast ratio: 1.248:1 (`#E6E6E6` / `#FFFFFF`)
  - **How to address**: add a visible border on hover. Use the border to avoid needing to drastically increase the contrast of the surface of the component (which would likely cause text contrast issues). Make sure to check the contrast between the border and the inner surface, as well as, the border and outer surface (i.e. the site header’s background).
- [ ] Idle state for “Search packages” field.
  - Border-to-inner surface:
    - Dark theme contrast 1.196:1 (`#262626` / `#161616`)
    - Light theme contrast ratio: 1.483:1 (`#CECECE` / `#F8F8F9`)
  - Border-to-outer surface (e.g. page background):
    - Dark theme contrast ratio: 1.257:1 (`#262626` / `#101010`)
    - Light theme contrast 1.574:1 (`#CECECE` / `#FFFFFF`)

## Keep the exist design as a non-default dimmed theme

There was some good design intent on creating a theme that was dimmed and would avoid issues with folks who have a hard time with higher contrast dark themes. I don’t think we need to discard this work as it does serve a particular accessibility need, it just shouldn’t be the default.
